src/app/domain/maven/MavenArtifact.php

```diff
@@ -164,23 +164,31 @@ private function getBaseUrlFromGroupIdAndArtifactId($targetGroupId, $artifactId)
   public function getVersions(): array
   {
     if ($this->versionCache == null) {
-      $baseUrl = $this->getBaseUrl();
-
-      $xml = HttpRequester::request("$baseUrl/maven-metadata.xml");
-
-      if (empty($xml)) {
-        $this->versionCache = [];
-        return $this->versionCache;
+      foreach ($this->archivedArtifacts as $archivedArtifact) {
+        $archivedArtifactBaseUrl = $this->getBaseUrlFromGroupIdAndArtifactId($archivedArtifact->getGroupId(), $archivedArtifact->getArtifactId());
```

Comment:
I would prefer to see a method which calculates the value for $this->versionCache